apps/gitness/src/framework/context/ThemeContext.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react'
+
+import { create } from 'zustand'
+import { persist } from 'zustand/middleware'
+
+import { ColorType, ContrastType, FullTheme, ModeType, ThemeState } from '@harnessio/ui/components'
+
+const useThemeStore = create<ThemeState>()(
+  persist(
+    set => ({
+      theme: 'system-std-std' as FullTheme, // Default theme
+      setTheme: (newTheme: FullTheme) => set({ theme: newTheme })
+    }),
+    {
+      name: 'canary-ui-theme' // LocalStorage key
+    }
+  )
+)
+
+export const useTheme = () => {
+  const { theme, setTheme } = useThemeStore(state => ({ theme: state.theme, setTheme: state.setTheme }))
+  return { theme, setTheme }
+}
+
+export function ThemeProvider({ children }: { children: React.ReactNode }) {
+  const { theme, setTheme } = useThemeStore(state => ({
+    theme: state.theme,
+    setTheme: state.setTheme
+  }))
+
+  useEffect(() => {
+    const root = window.document.documentElement
+
+    const getMediaQuery = () => window.matchMedia('(prefers-color-scheme: dark)')
```

Comment:
This will return a new media query each time called and it's called 3 times. Because it's new on each call, it'll not be removed on unmount. Not an issue in practice, but the fix is just to assign to a variable here instead of making it a function you always call.
```suggestion
    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)')
```

Reply:
Ahh got it, good to know! I'll fix it.